lib/internal/errors.js

```diff
@@ -1182,6 +1182,21 @@ E('ERR_UNKNOWN_ENCODING', 'Unknown encoding: %s', TypeError);
 E('ERR_UNKNOWN_FILE_EXTENSION', 'Unknown file extension: %s', TypeError);
 E('ERR_UNKNOWN_MODULE_FORMAT', 'Unknown module format: %s', RangeError);
 E('ERR_UNKNOWN_SIGNAL', 'Unknown signal: %s', TypeError);
+E('ESM_CJS_LOAD_ERROR', (error) => {
+
+  if(error instanceof SyntaxError && (/^import/mi).test(error.stack)){
```

Comment:
So this detection would fail code like `"  import 'sdf'"` - while rare, I think we should try and be as comprehensive as possible.

It also doesn't seem to support `export default 'sdf'` or code like `export { x } from './sdf'` etc.

The **export** cases are easy to handle because we can directly check:

```js
e.message.startsWith('SyntaxError: Unexpected token export')
```

So the import regular expression check as above is really just the super tough one to think about!

I can try throw together some ideas for this too, but perhaps lets discuss this on the next review further, rather than getting bogged down in it now - let's focus on getting everything else right first I think then come back around on this.

Reply:
what do you think about the `/^(\s|\t)*import/gmi` https://regexr.com/4i4gq ?